core/txbuilder/witness_test.go

```diff
@@ -41,9 +41,10 @@ func TestInferConstraints(t *testing.T) {
 }
 
 func TestWitnessJSON(t *testing.T) {
+	assetID := bc.NewAssetID([32]byte{0xff})
 	si := &SigningInstruction{
 		AssetAmount: bc.AssetAmount{
-			AssetID: bc.NewAssetID([32]byte{0xff}),
+			AssetId: &assetID,
```

Comment:
Yeah, I went back and forth on this. I'm similarly unsure that where I ended up is right, but as you say we can revisit this later.